worldedit-core/src/main/java/com/sk89q/worldedit/internal/anvil/ChunkDeleter.java

```diff
@@ -0,0 +1,327 @@
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.sk89q.worldedit.internal.anvil;
+
+import com.google.common.collect.Streams;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import com.sk89q.worldedit.math.BlockVector2;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.regions.CuboidRegion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiPredicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class ChunkDeleter {
+
+    public static final String DELCHUNKS_FILE_NAME = "delete_chunks.json";
+    private static final Logger logger = LoggerFactory.getLogger(ChunkDeleter.class);
+
+    private static final Comparator<BlockVector2> chunkSorter = Comparator.comparing(
+            pos -> (pos.getBlockX() & 31) + (pos.getBlockZ() & 31) * 32);
+
+    private static Gson chunkDeleterGson = new GsonBuilder()
+            .registerTypeAdapter(BlockVector2.class, new BlockVector2Adapter())
+            .setPrettyPrinting()
+            .create();
+
+    public static ChunkDeletionInfo readInfo(Path chunkFile) throws IOException, JsonSyntaxException {
+        String json = new String(Files.readAllBytes(chunkFile), StandardCharsets.UTF_8);
+        return chunkDeleterGson.fromJson(json, ChunkDeletionInfo.class);
+    }
+
+    public static void writeInfo(ChunkDeletionInfo info, Path chunkFile) throws IOException, JsonIOException {
+        String json = chunkDeleterGson.toJson(info, new TypeToken<ChunkDeletionInfo>() {}.getType());
+        try (BufferedWriter writer = Files.newBufferedWriter(chunkFile, StandardOpenOption.CREATE)) {
```

Comment:
Currently this will make the command fail with "File already exists", which doesn't really mean anything to the user. Perhaps we can check if the file exists or catch the appropriate exception, and inform the user that "There is already a scheduled chunk deletion"? Or merge the two deletions?

Reply:
It should (and from my testing, does) overwrite the current contents of the file with the new contents (which is any previously existing batches + the new batch).